src/core/linalg/src/dense/4C_linalg_tensor.hpp

```diff
@@ -45,28 +45,44 @@ namespace Core::LinAlg
   concept SquareTensor =
       Rank2TensorConcept<Tensor> && Tensor::template extent<0>() == Tensor::template extent<1>();
 
+
+  template <std::size_t... n>
+  struct NoCompression
+  {
+    static constexpr std::size_t compressed_size = (n * ...);
+
+    template <Internal::TensorBoundCheck bound_check>
+    static constexpr std::size_t flatten_index(decltype(n)... i)
+    {
+      return Internal::get_flat_index<Internal::OrderType::column_major, bound_check, n...>(i...);
+    }
+    // flatten index
+  };
   /*!
    * @brief An owning, dense tensor of arbitrary rank
    *
    * @copydetails TensorInternal
    */
   template <typename T, std::size_t... n>
-  using Tensor = TensorInternal<T, TensorStorageType::owning, n...>;  // owns the memory
+  using Tensor = TensorInternal<T, TensorStorageType::owning, NoCompression<n...>,
+      n...>;  // owns the memory
 
   /*!
    * @brief A dense view to a Tensor of arbitrary rank
    *
    * @copydetails TensorInternal
    */
   template <typename T, std::size_t... n>
-  using TensorView = TensorInternal<T, TensorStorageType::view, n...>;  // view onto's other
-                                                                        // memory
+  using TensorView = TensorInternal<T, TensorStorageType::view, NoCompression<n...>,
+      n...>;  // view onto's other
+              // memory
 
   /*!
    * @brief Creates a TensorView from a pointer to data and the tensor shape
    *
-   * This function creates a TensorView from a pointer to data and the tensor shape specified by the
-   * template parameters. The data is expected to be stored in a contiguous memory block of size
+   * This function creates a TensorView from a pointer to data and the tensor shape specified by
+   * the template parameters. The data is expected to be stored in a contiguous memory block of
+   * size
    * @p (n*...) with column-major layout.
```

Comment:
I am not sure this fold expression is understandable to people reading the docs.